recipes/hazelcast-cpp-client/4.0.0/conanfile.py

```diff
@@ -0,0 +1,86 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class HazelcastCxx(ConanFile):
+    name = "hazelcast-cpp-client"
+    description = "C++ client library for Hazelcast in-memory database."
+    license = "Apache-2.0"
+    topics = ("conan", "hazelcast", "client", "database", "cache")
+    homepage = "https://github.com/hazelcast/hazelcast-cpp-client"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": False
```

Comment:
Let's set this to true to check that it works.

There is suspicious code in cmake, which might not work.

https://github.com/hazelcast/hazelcast-cpp-client/blob/027d56df46d770d8e2225693e4bbc91703451757/CMakeLists.txt#L89

Reply:
Its a really common bug https://github.com/Thalhammer/jwt-cpp/pull/128/files#diff-8bc7d2f2d308dd6233c8a479ffa5239250be2ce47cb4014d7af34ec0c355c8dcR26

It might still work https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindOpenSSL.cmake#L177

Reply:
@mathbunnyru  But I don't want users by default use the openssl version since it means and extra dependency. But I sure test it locally with the option and it works but the CI environment is probably different of course. Is there a way to trigger CI tests with this option set to ON but keep the default OFF as it is now?

Reply:
What is the default of the project? We should follow that!

Reply:
Defualt is `with_openssl=False`